### PR TITLE
8 MachineService interface

### LINKED ISSUE #8: MachineService interface
**Description of the feature to be developed:**
Make the interface "MachineService" and add the abstract methods found in the class diagram to it.

### PR DESCRIPTION
**Pull request by Teyson**
Closes #8

## Description
Adds the MachineService interface

## Checklist:
- [x] My code follows the code style of this project.
- [ ] I have added tests to cover my changes, if necessary.
- [x] All tests pass (existing and potential new).
- [x] Sonar check has passed
- [x] No codesmells or bugs

## Other Relevant Information
No relevant tests.

### CODE REVIEW

File: BrewMesModulesParent/common/src/main/java/com/brewmes/common/services/IMachineService.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+    public Map<Integer, Connection> getConnections();
+
+    public boolean addConnection(String ip, int id);
+
+    public boolean removeConnection(int id);
+
+    public void startAutoBrew(int machineId);
+
+    public void stopAutoBrew(int machineId);
```
Comment (Lawst96) [inline BrewMesModulesParent/common/src/main/java/com/brewmes/common/services/IMachineService.java:24]: public is redundant in interfaces.

**Review by Lawst96 — CHANGES_REQUESTED** [review]
Give methods a javadoc comment, to describe desired functionality.

**Review by Wylardt — APPROVED** [review]

File: BrewMesModulesParent/common/src/main/java/com/brewmes/common/services/IMachineService.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+import com.brewmes.common.entities.Connection;
+import com.brewmes.common.util.Command;
+import com.brewmes.common.util.Products;
+
+import java.util.List;
+
+public interface IMachineService {
+
+    /**
+     * Takes a command and send it to the desired machine
```
Comment (Lawst96) [inline BrewMesModulesParent/common/src/main/java/com/brewmes/common/services/IMachineService.java:12]:
```suggestion
     * Takes a command and sends it to the desired machine
```

**Review by Kaszz — APPROVED** [review]

File: BrewMesModulesParent/common/src/main/java/com/brewmes/common/services/IMachineService.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+import java.util.List;
+
+public interface IMachineService {
+
+    /**
+     * Takes a command and sends it to the desired machine
+     * @param command The command that should be sent to the machine
+     * @param machineId The ID of the machine that should receive the command
+     */
+    void controlMachine(Command command, String machineId);
```
Comment (Lawst96) [inline BrewMesModulesParent/common/src/main/java/com/brewmes/common/services/IMachineService.java:16]:
```suggestion
    boolean controlMachine(Command command, String machineId);
```
Should probably return a response if the command was successful or not.

**Review by Kechr — APPROVED** [review]

File: BrewMesModulesParent/common/src/main/java/com/brewmes/common/services/IMachineService.java [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+    /**
+     * Sets the desired values for a production
+     * @param speed The desired speed that the production should run at
+     * @param beerType The desired beer type of the production
+     * @param batchSize The desired amount of beer to be produced
+     * @param machineId The ID of the desired machine the values should be set on
+     */
+    void setVariables(double speed, Products beerType, int batchSize, String machineId);
```
Comment (Lawst96) [inline BrewMesModulesParent/common/src/main/java/com/brewmes/common/services/IMachineService.java:25]:
```suggestion
    boolean setVariables(double speed, Products beerType, int batchSize, String machineId);
```
Should probably return a response if the command was successful or not.

File: BrewMesModulesParent/common/src/main/java/com/brewmes/common/services/IMachineService.java [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+    /**
+     * Sets up the desired subscriptions to a machines values.
+     * @param machineId The ID of the machine desired for the subscription
+     */
+    void subscribeToMachineValues(String machineId);
```
Comment (Lawst96) [inline BrewMesModulesParent/common/src/main/java/com/brewmes/common/services/IMachineService.java:31]:
```suggestion
    boolean subscribeToMachineValues(String machineId);
```
Should probably return a response if the subscription was successfully set up or not.

File: BrewMesModulesParent/common/src/main/java/com/brewmes/common/services/IMachineService.java [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+     * Gets all the connections from the database
+     * @return The list of Connection objects gotten from the database
+     */
+    List<Connection> getConnections();
+
+    /**
+     * Adds a connection to a Machine
+     * @param ip The IP address of the machine
+     * @param name The desired name of the machine, used for easy identification of the machine
+     * @return Returns true if the connection was successfully made, returns false if the connection failed.
```
Comment (Lawst96) [inline BrewMesModulesParent/common/src/main/java/com/brewmes/common/services/IMachineService.java:43]:
```suggestion
     * @return Returns true if the connection was successfully made, returns {@code false} if the connection failed.
```
Use `@code`.

> 

> 

>

File: BrewMesModulesParent/common/src/main/java/com/brewmes/common/services/IMachineService.java [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+     * @param machineId The id of the desired machine
+     * @return Returns true if the connection was successfully removed, returns false if the removal failed.
+     */
+    boolean removeConnection(String machineId);
+
+    /**
+     * Sets the machine in the autobrewing mode so it brews from the queue.
+     * @param machineId The id of the machine
+     */
+    void startAutoBrew(String machineId);
```
Comment (Lawst96) [inline BrewMesModulesParent/common/src/main/java/com/brewmes/common/services/IMachineService.java:58]:
```suggestion
    boolean startAutoBrew(String machineId);
```
Should probably return a response if the autobrew was successful.

File: BrewMesModulesParent/common/src/main/java/com/brewmes/common/services/IMachineService.java [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+
+    /**
+     * Stops the machine from autobrewing
+     * @param machineId The id of the machine
+     */
+    void stopAutoBrew(String machineId);
```
Comment (Lawst96) [inline BrewMesModulesParent/common/src/main/java/com/brewmes/common/services/IMachineService.java:64]:
```suggestion
    boolean stopAutoBrew(String machineId);
```
Should probably return a response if the stopping was successful or not.

**Review by Lawst96 — CHANGES_REQUESTED** [review]
Inconsistent use of 

> .

**Review by dottore1 — APPROVED** [review]

**Review by Lawst96 — APPROVED** [review]
:)

**Pull request merged**